turbinia/workers/artifact.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for running Plaso."""
+
+from __future__ import unicode_literals
+
+import os
+
+from turbinia import config
+from turbinia.evidence import ExportedFileArtifact
+from turbinia.workers import TurbiniaTask
+
+
+class FileArtifactExtractionTask(TurbiniaTask):
+  """Task to run image_export (log2timeline)."""
+
+  def __init__(self, artifact_name):
+    super(FileArtifactExtractionTask, self).__init__()
+    self.artifact_name = artifact_name
+
+  def run(self, evidence, result):
+    """Task that process data with Plaso.
+
+    Args:
+        evidence: Path to data to process.
+        result: A TurbiniaTaskResult object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    config.LoadConfig()
+
+    export_directory = os.path.join(self.output_dir, 'export')
+    image_export_log = os.path.join(self.output_dir, '{0:s}.log'.format(self.id))
+
+    cmd = [
+        'image_export.py',
+        '--logfile',
+        image_export_log,
+        '-w',
+        export_directory,
+        '--artifact_filters',
+        self.artifact_name,
+        ]
+    if config.DEBUG_TASKS:
+      cmd.append('-d')
+
+    # Path to the source image/directory.
+    cmd.append(evidence.local_path)
+
+    result.log('Running image_export as [{0:s}]'.format(' '.join(cmd)))
+
+    ret, _ = self.execute(cmd, result, save_files=[image_export_log])
+    if ret:
+      result.close(False, "image_export.py failed.")
```

Comment:
Small nit: I see both single and double quotes in this file.  Is there a convention, or can we make them all single quotes?

Reply:
Copy and paste from other bits of the codebase. This should have been caught by pylint though, I would have thought. Done.